chain/x/zoracle/handler_test.go

```diff
@@ -98,6 +97,94 @@ func TestReportInvalidValidator(t *testing.T) {
 	require.Equal(t, types.CodeInvalidValidator, got.Code)
 }
 
+func TestStoreCodeSuccess(t *testing.T) {
+	ctx, keeper := keep.CreateTestInput(t, false)
+	absPath, _ := filepath.Abs("../../wasm/res/test.wasm")
+	code, err := wasm.ReadBytes(absPath)
+	if err != nil {
+		fmt.Println(err)
+	}
+	owner := sdk.AccAddress([]byte("owner"))
+	codeHash := types.NewStoredCode(code, owner).GetCodeHash()
+
+	msg := types.NewMsgStoreCode(code, owner)
+	got := handleMsgStoreCode(ctx, keeper, msg)
+	require.True(t, got.IsOK(), "expected store code to be ok, got %v", got)
+
+	// Check codehash from event
+	require.Equal(t, types.EventTypeStoreCode, got.Events.ToABCIEvents()[0].Type)
+	require.Equal(t, hex.EncodeToString(codeHash), string(got.Events.ToABCIEvents()[0].Attributes[0].Value))
+
+	// Check value in store
+	sc, err := keeper.GetCode(ctx, codeHash)
+	require.Nil(t, err)
+
+	require.Equal(t, owner, sc.Owner)
+	require.Equal(t, code, sc.Code)
+}
+
+func TestStoreCodeFailed(t *testing.T) {
+	ctx, keeper := keep.CreateTestInput(t, false)
+	code := []byte("Code")
+
+	owner := sdk.AccAddress([]byte("owner"))
```

Comment:
Should we use realistic address here? (20 bytes?)

Reply:
Is it common in Cosmos-SDK tests to use "owner"?

Reply:
It's common thing e.g. `https://github.com/cosmos/cosmos-sdk/blob/master/x/bank/internal/keeper/keeper_test.go`